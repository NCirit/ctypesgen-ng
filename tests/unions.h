union simple {
	int x;
	float y;
};

union empty {
};

union nested {
	union simple s;
	long long z;
};

union forward_declaration;

union simple binary(union simple a, union simple b);

void binary2(union simple *a, const union simple *b);

union bitty {
	int a : 1;
	int b : 2;
	int c : 4;
	int d : 8;
	int e : 16;
};

struct tagged_union {
	int type;
	union simple u;
};
