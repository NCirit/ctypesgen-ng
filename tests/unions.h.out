import ctypes
import ctypes.util
import os


if not ('LIBRARY_PATH' in os.environ):
    if 'PYTHONPATH' in os.environ:
        os.environ['LIBRARY_PATH'] = os.environ['PYTHONPATH']
    else:
        os.environ['LIBRARY_PATH'] = os.path.dirname(os.path.abspath(__file__))

if os.name == "nt":
    if 'PYTHONPATH' in os.environ:
        os.environ['PATH'] += os.pathsep + os.environ['PYTHONPATH']
    else:
        os.environ['PATH'] += os.pathsep + os.path.dirname(os.path.abspath(__file__))
test_path = ctypes.util.find_library('test')
test = ctypes.CDLL(os.path.abspath(test_path))


class union_simple(ctypes.Union):
    pass
union_simple._pack_ = 1
union_simple._fields_ = [
    ('x', ctypes.c_int),
    ('y', ctypes.c_float),
]


class union_empty(ctypes.Union):
    pass


class union_nested(ctypes.Union):
    pass
union_nested._pack_ = 1
union_nested._fields_ = [
    ('s', union_simple),
    ('z', ctypes.c_longlong),
]


class union_forward_declaration(ctypes.Union):
    pass

test.binary.restype = union_simple
test.binary.argtypes = [union_simple, union_simple]

test.binary2.restype = None
test.binary2.argtypes = [ctypes.POINTER(union_simple), ctypes.POINTER(union_simple)]


class union_bitty(ctypes.Union):
    pass
union_bitty._pack_ = 1
union_bitty._fields_ = [
    ('a', ctypes.c_int, 1),
    ('b', ctypes.c_int, 2),
    ('c', ctypes.c_int, 4),
    ('d', ctypes.c_int, 8),
    ('e', ctypes.c_int, 16),
]


class struct_tagged_union(ctypes.Structure):
    pass
struct_tagged_union._pack_ = 1
struct_tagged_union._fields_ = [
    ('type', ctypes.c_int),
    ('u', union_simple),
]
